### PR TITLE
adds support for searching in integer and select fields

### PR DESCRIPTION
**Pull request by dodumosu**
this pull request adds support for querying submissions by field value (currently, **only integer and select fields are supported**). some screenshots to illustrate:

chrome_xSBqveIvTP
3P1RWCw3j5
QGBIYqaCQh
iwHQ1U4TBI

it supports multiple terms, but all terms are conjoined (that is, using `AND`).

### CODE REVIEW

**Comment by dodumosu** [comment]
this is both a PR and a request for feedback, especially as regards the UI and UX

**Comment by dodumosu** [comment]
chrome_WcsdSQ02Uw

chrome_cwK0UyjY4Z

added links from data summary

File: apollo/submissions/qa/query_builder.py [hunk 1]
```diff
@@ -200,6 +190,17 @@ def visit_comparison(self, node, children):
...
         return super().visit_comparison(node, children)
+    
+    def visit_exponent(self, node, children):
+        if len(children) == 1:
+            return children[0]
+
+        exponent = children[0]
+        for i in children[1:]:
+            # exponent **= i
+            exponent **= i
```
Comment (takinbo) [inline apollo/submissions/qa/query_builder.py:201]: What's the difference between `func.pow` and `**=`

**Review by takinbo — CHANGES_REQUESTED** [review]

File: apollo/submissions/qa/query_builder.py (hunk 1 above)
Comment (dodumosu) [reply apollo/submissions/qa/query_builder.py:201]: ah. okay, that's in the base class. `func.pow()` is only useful when we're generating a query. it won't work in the event that we're in the InlineQA subclass

**Review by takinbo — CHANGES_REQUESTED** [review]
Is there a better way to represent the value of the Fields Value textbox in the URL?

Currently, it looks like this: /submissions/165?participant_id=&sample=&location_group=&values=%5B%7B"value"%3A"AD%3D1"%7D%5D

which is not elegant. I will also send you a diff to apply to the tagify css to make the field more uniform with the rest of the controls.

**Comment by takinbo** [comment]
@dodumosu also use title case for the value "Field values". It should be "Field Values"

**Comment by takinbo** [comment]
Resolves nditech/apollo-issues#185

**Comment by dodumosu** [comment]
closing in favour of a new implementation which will be pushed soon.

**Comment by dodumosu** [comment]
replaced by #960

**Pull request closed without merging**